var/spack/repos/builtin/packages/brahma/package.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Brahma(CMakePackage):
+    """Interceptor library for I/O calls using Gotcha"""
+
+    homepage = "https://github.com/hariharan-devarajan/brahma"
+    git = "https://github.com/hariharan-devarajan/brahma.git"
+    maintainers("hariharan-devarajan")
+
+    version("develop", branch="dev")
+    version("master", branch="master")
+    version("0.0.1", tag="v0.0.1")
+
+    variant("mpi", default=False, description="Enable MPI support")
+    depends_on("cpp-logger@0.0.1:", when="@0.0.1")
+    depends_on("gotcha@develop", when="@0.0.1")
+    depends_on("catch2@3.0.1", when="@0.0.1")
```

Comment:
I recommend dropping these dependencies since there is no real difference given the dependencies apply for the listed version as well as the branches.

Otherwise, it would be equivalent to drop the directives without the `when` arguments and add a colon (`:`) at the end of the current `when` arguments (see https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison) to indicate the dependencies apply for the current version (tag) and newer (including `develop` and `master`).

Reply:
I removed the when directive for now.